testsuite/dynaconf_loader.py

```diff
@@ -53,17 +53,14 @@ def _guess_version(ocp):
     """Attempt to determine version from amp-system imagestream"""
 
     version = None
-    lookup = _docker_image(ocp, "amp-system")
+    config = ocp.do_action("get", ["dc/apicast-staging", "-o", "yaml"], parse_output=True)
+    lookup = [trigger.imageChangeParams["from"]["name"].split(":", 1)[1]
+              for trigger in config.model.spec.triggers if trigger.type == "ImageChange"][0]
 
     try:
-        version = lookup["name"]
-        Version(version)
+        Version(lookup)
     except (KeyError, InvalidVersion):
```

Comment:
if original code expected KeyError this should probably handle IndexError and try should cover one line more